src/hooks/useLikePost.js

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { dislikePost, likePost } from "../features";
+
+export const useLikePosts = (post) => {
+  const dispatch = useDispatch();
+  const [isLiked, setLiked] = useState(false);
+  const { user: authUser, token } = useSelector((store) => store.auth);
+  const { posts } = useSelector((store) => store.posts);
+
+  useEffect(() => {
+    (() => {
+      if (
+        Boolean(
+          post.likes.likedBy.find(
```

Comment:
try to use ?. after objects in case they return empty they won't break the screen.

Reply:
yes, I mostly use them where there's a possibility that the property won't be present